File: src/main/java/net/kigawa/kweb/controller/AuthController.java
package net.kigawa.kweb.controller;

import net.kigawa.kweb.bean.*;
import org.springframework.beans.factory.annotation.*;
import org.springframework.stereotype.*;
import org.springframework.ui.*;
import org.springframework.web.bind.annotation.*;

@Controller
public class AuthController
{
    private final URIUtil uriUtil;

    @Autowired
    public AuthController(URIUtil uriUtil)
    {
        this.uriUtil = uriUtil;
    }

    @RequestMapping(value = {
            "/test"
    })
    public String top(Model model)
    {
        model.addAttribute("baseurl",
                uriUtil.urlFromMapping("root"));
        return "index";
    }
}

File: src/main/java/net/kigawa/kweb/controller/AuthController.kt
package net.kigawa.kweb.controller

import net.kigawa.kweb.bean.URIUtil
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import javax.servlet.http.HttpServletResponse

@RestController
@RequestMapping(value = ["/auth"])
class AuthController @Autowired constructor(private val uriUtil: URIUtil) {
    @RequestMapping(value = ["/token"])
    fun token(@CookieValue("token", required = false) token: String?, response: HttpServletResponse): String {
        return "index"
    }
}